### DISCUSSION TITLE
socket.io-p2p, web-rtc, rtmp

### DISCUSSION THREAD

**Discussion by todaynogada** [feathersjs/feathers#2530] [Q&A] [answered]
Hello. I am a developer using feathersjs.
feathersjs I want to know whether the framework supports socket.io-p2p, web-rtc, rtmp, etc.
Also, the existing socket has to communicate with the server once, but I wonder if p2p socket communication can be implemented with this framework now.

**Comment by daffl** [comment, marked as answer]
There currently is no p2p transport mechanism (related issue: https://github.com/feathersjs/feathers/issues/1022) but it is something I am very much interested in and have been discussing with my friends at https://fission.codes. A while ago I also made a decentralized Feathers chat demo that uses the Beaker browser (slides for the talk are at: https://daffl.github.io/decentralized-web/).

The problem is that just enabling a low level peer-to-peer transport layer won't be enough to create a complete Feathers app since you will need several systems on top of that. Specifically:

- Distributed identity ([UCAN](https://fission.codes/blog/auth-without-backend/))
- Encrypted data storage (probably on top of [IPFS](https://ipfs.io/))
- A decentralized database
